posthog/hogql/constants.py

```diff
@@ -103,3 +107,11 @@
 # Never return more rows than this in top level HogQL SELECT statements
 DEFAULT_RETURNED_ROWS = 100
 MAX_SELECT_RETURNED_ROWS = 65535
+
+# Settings applied on top of all HogQL queries.
+class HogQLSettings(BaseModel):
+    class Config:
+        extra = Extra.forbid
+
+    readonly: Optional[int] = 1
```

Comment:
Do we want to be able to remove `readonly`? Should this be constant 1 forcing us to change code if we wanted to let HogQL do non-selects?

Reply:
So this is allowed

```
import dataclasses
from typing import Optional


@dataclasses.dataclass
class Thing:
    optional_value: Optional[int] = 1


x = Thing()
y = Thing(optional_value=None)

print(x.optional_value)
print(y.optional_value)
```

this prints 

```
1
None
```

So, if we can accidentally or someone can maliciously get us to create settings with `HogQLSettings(readonly=None)` then we don't have readonly set on the query.

If we're saying that HogQL is readonly a constant here feels safer

Reply:
Yeah, but none of the settings come from users --> we set them ourselves, and it's hardcoded in. I opted out of using feature flags to set these settings just for this reason. Now, if we want, we could, e.g. override some of them for some longer queries, e.g. setting max execution time to 5min for exports. 

Theoretically we can also override `readonly`, but 1) nothing does that now, 2) users will not be able to set these settings... at least not directly, 3) it's still a select query so it's anyway an extra safety measure.